SU2_CFD/src/drivers/CSinglezoneDriver.cpp

```diff
@@ -312,3 +312,83 @@ bool CSinglezoneDriver::Monitor(unsigned long TimeIter){
 bool CSinglezoneDriver::GetTimeConvergence() const{
   return output_container[ZONE_0]->GetCauchyCorrectedTimeConvergence(config_container[ZONE_0]);
 }
+
+void CSinglezoneDriver::PreRunSpectralRadius() {
+  CGeometry* geometry = geometry_container[ZONE_0][INST_0][MESH_0];
+  CVariable* nodes = nullptr;
+
+  // Get total number of variables
+  unsigned long nVarTotal = 0;
+  unsigned long nPoint = geometry->GetnPointDomain();
+  for (auto iSol = 0u; iSol < MAX_SOLS; ++iSol) {
+      CSolver* solver = solver_container[ZONE_0][INST_0][MESH_0][iSol];
+      if (solver) nVarTotal += solver->GetnVar();
+  }
+
+  unsigned long nTotal = nVarTotal * nPoint;
+
+  // Initialize power iteration vector
+  if (v_estimate.empty()) {
+      v_estimate.resize(nTotal);
+      passivedouble norm = 0.0;
+      for (unsigned long i = 0; i < nTotal; i++) {
+          v_estimate[i] = static_cast<passivedouble>(rand()) / RAND_MAX;
+          norm += v_estimate[i] * v_estimate[i];
+      }
+      norm = sqrt(norm);
+      for (unsigned long i = 0; i < nTotal; i++) v_estimate[i] /= norm;
+  }
+
+  // Seed derivatives for all solvers
+  SeedAllDerivatives(v_estimate, nodes, geometry);
```

Comment:
nodes is nullptr in these calls. You don't need to pass nodes here, the solver owns its nodes.